agent/dockerclient/dockerapi/errors.go

```diff
@@ -151,6 +153,12 @@ func (err CannotPullContainerError) ErrorName() string {
 	return "CannotPullContainerError"
 }
 
+func (err CannotPullContainerError) Constructor() func(string) apierrors.NamedError {
```

Comment:
maybe this is a bit unconventional. I believe in go we usually follow `NewSomething` naming convention. 

I wonder instead of adding this to `CannotPullContainerError` - should we do something like this:

```go
func NewCannotPullContainerError(string) 
```

Reply:
this is the implementation of an interface
```
type ConstructibleNamedError interface {
	NamedError
	Constructor() func(string) NamedError
}
```